### PR TITLE
Add a --no-empty flag to disallow empty env vars

### PR DESCRIPTION
**Pull request by Globegitter**
Just to have some stricter error checking when we know we do not want any empty env variables.

### CODE REVIEW

**Comment by relaxdiego** [comment]
Thanks @globegitter! (Awesome nick, BTW!)

Before I can merge this, can you add a test for this in test-render? You might need to add a 3 arg to the it() function where you can pass the --no-empty flag.

**Comment by Globegitter** [comment]
Done.

File: test-render [hunk 1]
```diff
@@ -74,6 +74,16 @@ somerandomtextmorerandomtext
 \$symbol_that_doesnt_get_rendered
 EOF
 
+it "does throw an error if test_variable is defined but empty if --no-empty flag is provided" "test_variable=" "--no-empty" <<EOF
+ERROR: test_variable is empty
+EOF
+
+it "doesn' throw an error if test_variable is defined but empty and --no-empty flag is provided on last position" "test_variable=" "" "--no-empty" <<EOF
```
Comment (relaxdiego) [inline test-render]: Thanks for adding the tests, @Globegitter! I was going to argue that it should work no matter where the flag is placed but thankfully I had the sense to review my POSIX and realized that you were right in doing this. :-)

Just one tiny nitpick though: Can you add the `t` to `doesn'` above?

**Review by relaxdiego — CHANGES_REQUESTED** [review]

**Review by relaxdiego — APPROVED** [review]

**Comment by relaxdiego** [comment]
Thanks @Globegitter! First PR merged. W00t! This deserves a beer! 🍻

**Comment by Globegitter** [comment]
Thank you - I have to say from all the simple templating solutions out there I liked yours ones the most - there is not many that kept it that simple :)
